tests/integration/components/tasks-test.js

```diff
@@ -80,4 +80,79 @@ module('Integration | Component | tasks', function (hooks) {
     );
     assert.dom('[data-test-task-spinner]').doesNotExist();
   });
+  test('changing the task status from any status other than BLOCKED to IN_PROGRESS inform proceeding further will make progress 0', async function (assert) {
+    tasks[0].status = 'SMOKE_TESTING';
+    this.set('dev', true);
+    const ctrl = this.owner.lookup('controller:tasks');
+
+    // before action
+    assert.equal(ctrl.showModal, false);
+    assert.equal(ctrl.message, '');
+
+    ctrl.allTasks = tasks;
+    ctrl.dev = true;
+    ctrl.taskFields = {
+      status: 'IN_PROGRESS',
+    };
+
+    //action
+    ctrl.send('handleUpdateTask', tasks[0].id, () => {});
+
+    //after action
+    assert.equal(ctrl.showModal, true);
+    assert.equal(
+      ctrl.message,
+      'The progress of current task is 25%. Proceeding further will make task progress 0%.'
+    );
+  });
+  test('changing the task status from IN_PROGRESS to any status other than BLOCKED inform proceeding further will make progress 100', async function (assert) {
+    tasks[0].status = 'IN_PROGRESS';
+    this.set('dev', true);
+    const ctrl = this.owner.lookup('controller:tasks');
+
+    // before action
+    assert.equal(ctrl.showModal, false);
+    assert.equal(ctrl.message, '');
+
+    ctrl.allTasks = tasks;
+    ctrl.dev = true;
+    ctrl.taskFields = {
+      status: 'SMOKE_TESTING',
+    };
+
+    // action
+    ctrl.send('handleUpdateTask', tasks[0].id, () => {});
+
+    // after action
+    assert.equal(ctrl.showModal, true);
+    assert.equal(
+      ctrl.message,
+      'The progress of current task is 25%. Proceeding further will make task progress 100%.'
+    );
+  });
+  test('changing the task status from BLOCKED to any status other than IN_PROGRESS when percentCompleted is less than 100, inform the user proceeding further will make progress 100', async function (assert) {
+    tasks[0].status = 'BLOCKED';
+    this.set('dev', true);
+    const ctrl = this.owner.lookup('controller:tasks');
+
+    // before action
+    assert.equal(ctrl.showModal, false);
+    assert.equal(ctrl.message, '');
+
+    ctrl.allTasks = tasks;
+    ctrl.dev = true;
+    ctrl.taskFields = {
+      status: 'SMOKE_TESTING',
+    };
+
+    // action
```

Comment:
Write unit tests to cover more scenarios, including edge cases and error conditions, to ensure the robustness and reliability of the code.